File: server/services/battle/wall_bonus.dart
class WallBonus<T> {

  const WallBonus(this.defBonus, this.def);

  factory WallBonus.off(T defBonus, T def) {
    return WallBonus(defBonus, def);
  }
  final T defBonus;
  final T def;
}

File: java_server/src/main/java/xyz/qruto/java_server/models/battle/WallBonus.java
package xyz.qruto.java_server.models.battle;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class WallBonus<T> {
    private T defBonus;
    private T def;

    public static <T> WallBonus<T> off(T defBonus, T def){
        return new WallBonus<>(defBonus, def);
    }
}
